end_to_end/instant_answer.py

```diff
@@ -16,34 +16,54 @@ def assert_autocomplete(query: str, doc: str):
     assert response.json()['data']['result']['places'][0]['id'] == doc
 
 
-def test_paris():
+def test_admin_paris():
     assert_autocomplete("http://localhost:5000/v1/instant_answer?q=paris", 'admin:osm:relation:71525')
 
 
-def test_hotel_gabriel_paris():
-    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=hotel gabriel paris", 'ta:poi:529918')
+def test_admin_bruges():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Bruges", 'admin:osm:relation:562654')
+
+
+def test_admin_la_balme():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=La Balme-de-Sillingy", 'admin:osm:relation:103823')
+
+
+def test_admin_megeve():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Megève", 'admin:osm:relation:75312')
+
+
+def test_admin_saffre():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Saffré", 'admin:osm:relation:173715')
 
 
 def test_mdph():
     assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Direction de la Solidarité Départementale - MDPH",
                         'osm:relation:1672143')
 
 
-def test_bruges():
-    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Bruges", 'admin:osm:relation:562654')
+def test_hotel_should_call_tripadvisor_when_hotel_category_is_detected():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=hotel gabriel paris", 'ta:poi:529918')
 
 
-# def test_mediateque():
-#     assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Médiathèque Jules Verne", 'osm:node:2870807394')
+def test_restaurant_with_france_place_intention_should_call_pagesjaunes_for_poi():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=garden bistro lyon", 'pj:58140933')
 
 
-def test_la_balme():
-    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=La Balme-de-Sillingy", 'admin:osm:relation:103823')
+def test_poi_without_place_intention_should_call_tripadvisor_when_found():
+    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=garden bistro", 'ta:poi:1509459')
 
 
-def test_megeve():
-    assert_autocomplete("http://localhost:5000/v1/instant_answer?q=Megève", 'admin:osm:relation:75312')
+def test_restaurant_without_place_intention_should_call_tripadvisor_when_found():
```

Comment:
I feel like in the future we should consider grouping some tests if naming them becomes too tedious :shrug: